mne/epochs.py

```diff
@@ -1342,6 +1342,8 @@ def _get_data(self, out=True, verbose=None):
                         epoch = self._data[idx]
                 else:  # from disk
                     epoch_noproj = self._get_epoch_from_raw(idx)
+                    if epoch_noproj is None:  # Dropped due to bad segment
+                        continue
```

Comment:
I agree with @Eric89GXL there is a cleaner way.

The logic of making an epoch bad should be in the _is_good_epoch method.

I don't have time to look into this now.

@Eric89GXL if you have time to take over otherwise I'll have a look later this week.